.github/workflows/ci.yml

```
@@ -13,10 +13,15 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-python@v4
+      - uses: actions/setup-python@v5
         with:
           python-version: "3"
-          cache: pip
+      - name: Install uv
+        run: >
+          curl --no-progress-meter --location --fail
+          --proto '=https' --tlsv1.2
+          "https://astral.sh/uv/install.sh"
+          | sh
```

I think it's less complex? (or the complexity is hidden ;) aka managed for us)

```suggestion
      - name: Install uv
        uses: hynek/setup-cached-uv@v2
```

---

*Explicit is better than implicit*! More to the point, `hynek/setup-cached-uv@v2` fetches from a third-party domain without being clear about it, which I'm not a massive fan of (that action also doesn't have the proto or tls flags which protect against downgrade attacks). We could change the `curl` command here to the suggested `curl -LsSf https://astral.sh/uv/install.sh | sh`, but I think it is better to use long flags (`-sS ~= --no-progress-meter`, `-L == --location`, `-f == --fail`) and have the downgrade protection.

Taking the following commits as examples, there seems to be a c. 0.07s-0.1s speed-up in the resolving step with the action and a 0.04s-0.06s speed-up in the preparing step.

Personally, I would rather be explicit about the install (especially as we are piping-to-sh installing a third-party script) than loose the ~0.15s speed-up. But perhaps having the complexity hidden is worth it on its own.

----

beb7956 (baseline, initialises cache)
```
Using Python 3.12.4 interpreter at: /opt/hostedtoolcache/Python/3.12.4/x64/bin/python3
Creating virtualenv at: ./venv
Resolved 43 packages in 195ms
Prepared 43 packages in 480ms
Installed 43 packages in 49ms
```

2f19ac5 (first with cache)
```
Using Python 3.12.4 interpreter at: /opt/hostedtoolcache/Python/3.12.4/x64/bin/python3
Creating virtualenv at: ./venv
Resolved 43 packages in 127ms
Prepared 43 packages in 418ms
Installed 43 packages in 47ms
```

ad74301 (empty commit)
```
Using Python 3.12.4 interpreter at: /opt/hostedtoolcache/Python/3.12.4/x64/bin/python3
Creating virtualenv at: ./venv
Resolved 43 packages in 95ms
Prepared 43 packages in 437ms
Installed 43 packages in 59ms
```

---

I'm fine with using it, it's maintained by a core developer who also maintains lots of popular packages, so I trust care is being taken.

It would be nice to outsource uv installation: there's a bunch of options for that curl command that might need changing later.

Plus this workflow is not creating any build artifacts used elsewhere: we're building docs, checking links, and throwing the whole environment away.

---

Fair enough!